### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by caspiansusskind**
README for the entire project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,59 +1,64 @@
...
+  - [Run Code](#run-code)
 
-### Linux & Mac OS
 
-Before proceding it is recommended to install the `libasound-dev` system package (or `portaudio` for Mac OS), since `PyAudio` cannot compile without it.
+## Architecture
+The architecture can be divided into two different parts: the *API layer*, which consists of functionality for handling data, GUI, ML/AL etc. - and the *execution layer*, which is responsible for utilizing the API to perform required tasks. All code relating to the API may be found in `src/api/` while the execution code lies in `src/`.
 
-Once this is done, the following steps should be performed:
+More specifically, the execution layer is devided into 5 different so called *processes*. When the program starts, one process is executed, and when the process terminates, so does the program. In order to communicate between processes, files are usually outputted so that they can be read between each other.
```
Comment (Ludvig-Joborn) [inline README.md:18]: divided

File: README.md [hunk 2]
```diff
@@ -1,59 +1,64 @@
...
...lity to add a *sliding window* of a user defined size to a data set exists. This is in order to try to capture the chronological correlation of emotions in music, and it can be done in a few different ways. In the implementation a separation is made between how training and test data is configured with a sliding window. For training data, the labels from the training data set are used as indicators of emotion for the previous samples, while for test data a basic linear regressor is used to predict the emotion of previous samples in order to get an indication of the emotion. For both training data and test data a prior must be chosen, the prior will act as labels/predictions for the first samples of each song in the train/test dataset when configuring the sliding window. The options for training data are, one prior for all songs and no prior at all. When no prior is passed, the labels from the first few samples are used and those samples are then not used during the training. The options for test data are either one prior for all songs or a different prior for each song. Below is an illustration of how a sliding window size of *r* works on one song in the training data with a prior.
```
Comment (Ludvig-Joborn) [inline README.md:50]: "... the labels from the first few samples are used and those samples are then not used during the training"
Litet förvirrande mening.

File: docs/dataset.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+**The features will be outputted as:** \
+`data/feature_extraction_<datetime>.csv`
+
+
+## Feature Selection
+Two types of feature selection methods are available at the moment; Principal Component Analysis (PCA) and Variance Threshold (VT).
+
+More information can be found here:
+- [PCA]([link](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html))
```
Comment (Ludvig-Joborn) [inline docs/dataset.md:83]: Länk funkar inte.

**Review by Ludvig-Joborn — CHANGES_REQUESTED** [review]
Fix these and we're done.

**Review by Ludvig-Joborn — APPROVED** [review]
